packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/send/dq-requests.ts

```diff
@@ -15,73 +15,71 @@ export type DQSamlClientResponse = SamlClientResponse & {
   outboundRequest: OutboundDocumentQueryReq;
 };
 
-export async function sendSignedDQRequests({
-  signedRequests,
+export async function sendSignedDqRequest({
+  request,
   samlCertsAndKeys,
   patientId,
   cxId,
+  index,
 }: {
-  signedRequests: BulkSignedDQ[];
+  request: SignedDqRequest;
   samlCertsAndKeys: SamlCertsAndKeys;
   patientId: string;
   cxId: string;
-}): Promise<DQSamlClientResponse[]> {
+  index: number;
+}): Promise<DQSamlClientResponse> {
   const trustedKeyStore = await getTrustedKeyStore();
-  const requestPromises = signedRequests.map(async (request, index) => {
```

Comment:
instead of passing in a bunch of requests, we just take one request at a time

Reply:
Sounds good.

It will increase the number of times this gets executed, so maybe we can externalize the retrieving of the trusted key store? It doesn't get any parameter and, since this is run on a lambda, its a quick win to move to a global variable in the file - loaded when the NodeJS environment loads the file, roughly once per lambda execution.

Reply:
yup sounds good

Reply:
getting the trusted key store is async so I had to move it into the ihe gateway logic module where it will similarily be called once per lambda execution

Reply:
I see. We could have it "cached" on the local file/module, though, something like:

```ts
let trustedStore: string | undefined = undefined;
function getTrustedKeyStore(): Promise<string> {
  if (!trustedStore) trustedStore = await originalGetTrustedKeyStore();
  return trustedStore;
}
```

Then we could keep as you had originally (calling `getTrustedKeyStore` inside `sendSignedDqRequest`) and still avoid loading the thing for every request.

The issue w/ requiring it as a parameter is that it doesn't seem like its the clients' responsibility to know about or deal with/figure out.? It looks like we're exposing an implementation detail.